src/wasm/literal.cpp

```diff
@@ -47,6 +51,19 @@ Literal::Literal(const uint8_t init[16]) : type(Type::v128) {
   memcpy(&v128, init, 16);
 }
 
+Literal::Literal(std::shared_ptr<GCData> gcData, Type type)
+  : gcData(gcData), type(type) {
+  // Null data is only allowed if nullable.
+  assert(gcData || type.isNullable());
+  // The type must be a proper type for GC data.
+  assert(isGCData());
+}
+
+Literal::Literal(RttSupers rttSupers, Type type)
```

Comment:
Do we pass this by value?

Reply:
We did, but I just changed it following @tlively 's suggestion to use move semantics.